plugins/parcelize/parcelize-compiler/testData/codegen/simplePersistentList.asm.ir.txt

```diff
@@ -0,0 +1,48 @@
+public final class Test$Creator : java/lang/Object, android/os/Parcelable$Creator {
+    public void <init>()
+
+    public final Test createFromParcel(android.os.Parcel parcel) {
+        LABEL (L0)
+          ALOAD (1)
+          LDC (parcel)
+          INVOKESTATIC (kotlin/jvm/internal/Intrinsics, checkNotNullParameter, (Ljava/lang/Object;Ljava/lang/String;)V)
+          NEW (Test)
+          DUP
+          ALOAD (1)
+          INVOKEVIRTUAL (android/os/Parcel, createStringArrayList, ()Ljava/util/ArrayList;)
+          CHECKCAST (java/lang/Iterable)
+          INVOKESTATIC (kotlinx/collections/immutable/ExtensionsKt, toPersistentList, (Ljava/lang/Iterable;)Lkotlinx/collections/immutable/PersistentList;)
```

Comment:
kotlinx.collections.immutable is not a library with a stable API surface. What is your plan in case its API/ABI changes?

Reply:
That's a good point. If the API changes we would have to find a way to support multiple versions of the library by dynamically checking what is available in the version being included in the compilation. That can indeed get messy (which is why only builtin library types are supported by parcelize at this point in time).

Are there any stability plans for `kotlinx.collections.immutable`? Compose is using the types to infer stability (which as long as types with these names in this package stay immutable is not a problem - if something changes name it just becomes slower with compose until the compose compiler plugin is updated, but it will still work.) 

Based on the amount of interest in getting these immutable collections supported for parcelize, Compose is driving a good amount of adoption.

Reply:
I think that if there are API changes, they will most likely be done through deprecation. So, we will definitely have time to support the new APIs. Also, I'm not sure there will be any reason to change the particular APIs we depend on, since they seem pretty natural.